### PR TITLE
feat(muta): link up metadata service with muta

### PR DESCRIPTION
**Pull request by rev-chaos**
**What type of PR is this?**
feat

**What this PR does / why we need it**:
link up metadata with muta.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: devtools/chain/genesis.toml [hunk 1]
```diff
@@ -4,3 +4,7 @@ prevhash = "44915be5b6c20b0678cf05fcddbbaa832e25d7e6ac538784cd5c24de00d47472"
 [[services]]
 name = "asset"
 payload = '{ "id": "f56924db538e77bb5951eb5ff0d02b88983c49c45eea30e8ae3e7234b311436c", "name": "MutaToken", "symbol": "MT", "supply": 320000011, "issuer": "f8389d774afdad8755ef8e629e5a154fddc6325a" }'
+
+[[services]]
+name = "metadata"
+payload = '{ "chain_id": "b6a4d7da21443f5e816e8700eea87610e6d769657d6b8ec73028457bf2ca4036", "common_ref": "703873635a6b51513451", "timeout_gap": 20, "cycles_limit": 99999999, "cycles_price": 1, "interval": 3000, "verifier_list": [{"address": "f8389d774afdad8755ef8e629e5a154fddc6325a", "propose_weight": 1, "vote_weight": 1}], "propose_ratio": 15, "prevote_ratio": 10, "precommit_ratio": 10}'
```
Comment (yejiayu) [inline devtools/chain/genesis.toml]: new line

File: protocol/src/types/genesis.rs [hunk 2]
```diff
@@ -7,6 +7,17 @@ pub struct Genesis {
     pub services:  Vec<ServiceParam>,
 }
 
+impl Genesis {
+    pub fn get_payload(&self, name: &str) -> &str {
+        &self
+            .services
+            .iter()
+            .find(|&service| service.name == name)
+            .expect("miss metadata service!")
```
Comment (yejiayu) [inline protocol/src/types/genesis.rs]: miss "name" service?

File: src/default_start.rs [hunk 3]
```diff
@@ -155,16 +148,49 @@ pub async fn start<Mapping: 'static + ServiceMapping>(
...
+        Arc::clone(&mempool),
+        Arc::clone(&storage),
+        Arc::clone(&trie_db),
+        Arc::clone(&service_mapping),
+    );
+
+    let exec_resp = futures::executor::block_on(api_adapter.query_service(
+        Context::new(),
+        current_block.header.height,
+        99999,
```
Comment (yejiayu) [inline src/default_start.rs]: u64 max

**Comment by yejiayu** [comment]
Missing code to read the metadata after each block execution and update it in `consensus.status`.

**Comment by rev-chaos** [comment]
> Missing code to read the metadata after each block execution and update it in `consensus.status`.

plan to do it in next PR

**Pull request merged**